enhancements/installer/compact-plus-worker.md

```diff
@@ -0,0 +1,194 @@
+  handle workloads that require replication.
+
+
+### API Extensions
+
+- Sets the `infrastructureTopology` field to the value of`ControlPlaneTopology` in the
+  `config.openshift.io/v1/infrastructure` CR called `cluster`, which is created
+  during the install process, when the number of workers is 1.
+
+- Sets the `mastersSchedulable` field to true in the
+  `config.openshift.io/v1/scheduler` CR called `cluster`, which is created
+  during the install process, when the number of workers is 1.
+
+### Implementation Details/Notes/Constraints [optional]
+
+What are the caveats to the implementation? What are some important details that
+didn't come across above. Go in to as much detail as necessary here. This might
+be a good place to talk about core concepts and how they relate.
+
+### Risks and Mitigations
+
+This should not introduce newer risks other than the existing ones in a compact
+cluster deployment.
+
+## Design Details
+
+### Open Questions [optional]
+
+Nothing at the moment
+
+### Test Plan
+
+
+- Add unit tests to ensure the Infrastructure and Scheduler settings are properly set for the different combinations
+
+- Add a periodic job (nightly?) job to run OpenShift tests on a 4-node cluster
+
+### Graduation Criteria
+
+**Note:** *Section not required until targeted at a release.*
+
+TBD
+
+#### Dev Preview -> Tech Preview
+
+TBD
+
+#### Tech Preview -> GA
+
+TBD
+
+#### Removing a deprecated feature
+
+TBD
+
+### Upgrade / Downgrade Strategy
+
+This enhancement does not have an impact on existing clusters as it's intended
+to affect only day0 operations.
+
+### Version Skew Strategy
+
+Does not apply
+
+### Operational Aspects of API Extensions
+
+
+- Allows for a 4-nodes, highly available, cluster to be deployed as a day0
+  task. It reduces the time required for such deployments to be created.
+
+- Does not impact general deployment functionalities in a negative way.
+
+- Allowing workloads on the master nodes may affect the master's health if the
+  traffic and workloads are too high.
+
+#### Failure Modes
+
+Same as a compact cluster
+
+#### Support Procedures
+
+- The same support procedures used for a compact cluster should be used for a
+  4-nodes cluster. I would appreciate help filling in this section with more
+  details.
+
+
+## Implementation History
+
+N/A
+
+## Drawbacks
+
+A 4-nodes cluster (a.k.a 3+1) is generally not a recommended architecture. By
+having schedulable masters users will be routing ingress traffic through the
+masters as well as other workloads. Depending on the amount of traffic, and the
+resources available to the cluster, this may result in degraded functionality
+in the masters, which finally affects the general health of the cluster.
+
+With this enhancement, we will be making it easier for users to create such
+4-nodes cluster deployments, which could make users believe it is supported or
+recommended.
+
+## Alternatives
+
+- Don't set masters as schedulable automatically. Instead, cluster
```

Comment:
I think it may be worth it to elaborate on the pros/cons of this alternate proposal. 

One of the pros is certainly that it will let the user be explicit about their desired control plane topology. However, this may result in the user still being able to deploy a 3+1 topology like today where the infrastructure topology is SingleReplica. 

Maybe some questions to add to the open questions section could be:

- Do we want to protect the user from having infra workloads that are not HA?
- Do we want the user to decide whether their infra workload should be HA or not?

Reply:
Removing the automatic scheduling behavior would also be a backwards-incompatible change, right?

Reply:
> Removing the automatic scheduling behavior would also be a backwards-incompatible change, right?

I meant that we could keep the current behavior (make masters schedulable if there are no workers) and not expand as proposed in this enhancement.